### ISSUE TITLE
Anti-Armor Modifier on ProjectE Dark Matter causes kills to not come from the player.

### ISSUE THREAD

**Issue by etmoonshade** [ianm1647/bettercompat#1]
... I think. :)

Version: bettercompat-0.4.1-1.16.5-36.2.20

tl;dr version to reproduce:
1. Load a red matter sword. Load a dark matter sword (or construct a sword with dark matter and a sword with anything BUT dark matter)
2. Attack a mob, kill it.
3. Note that attacks with a dark matter sword don't have the "sweep" attack animation, while attacks with a red matter sword do.
4. If you've got an enderman farm (43 block drop, etc.,) note the difference in drop rates between attacks with the dark matter sword and attacks with the other sword.

What I observed in-game:
I had a weapon with severing VII, luck III, and a couple of other modifiers. This should have basically given me a 100% chance to drop heads. Instead, I got ZERO drops when killing blazes (and it took me a shamefully long time to realize it <_<)

I finally sat down and played around with various combinations - first, a dark matter blade and red matter other stuff, then finally I just loaded one sword each of dark and red matter to compare. The dark matter sword didn't have that sweeping attack animation, while the red matter sword did. And hitting mobs in my enderman farm, the red matter sword dropped plenty of XP+pearls+other moddy stuff, while the dark matter sword dropped 2 pearls in probably 10-15 kills.

My suspicion:
https://github.com/ianm1647/bettercompat/blob/3e844bb2db2e28b2b3900b1797242474a1f2ca5a/src/main/java/com/bettercompat/main/modifiers/trait/AntiArmorModifier.java line 22:

> target.attackEntityFrom(DamageSource.GENERIC.setDamageBypassesArmor(), damage + boost);

It almost looks like this is rewriting the damage source from the player to (functionally) nothing? I know that if I just let endermen drop to their deaths, I get fewer drops overall.

Hope that information is enough to point you in the right direction. I'll keep an eye out for replies. :)

**Comment by etmoonshade** [comment]
uh. Also, with the opening of this issue, _apparently_ the link on the mod page points to this bug report. :D

I guess there was supposed to be a list of potential mods to be compatible with?

**Comment by ianm1647** [comment]
this is late but im fixing this currently
